pandas/tests/arrays/masked/test_arrow_compat.py

```diff
@@ -89,12 +86,89 @@ def test_arrow_sliced(data):
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.fixture
+def np_dtype_to_arrays(any_real_dtype):
+    np_dtype = np.dtype(any_real_dtype)
+    pa_type = pa.from_numpy_dtype(np_dtype)
+
+    # None ensures the creation of a bitmask buffer.
+    pa_array = pa.array([0, 1, 2, None], type=pa_type)
+    # Since masked Arrow buffer slots are not required to contain a specific
+    # value, assert only the first three values of the created np.array
+    np_expected = np.array([0, 1, 2], dtype=np_dtype)
+    mask_expected = np.array([True, True, True, False])
+    return np_dtype, pa_array, np_expected, mask_expected
+
+
+def test_pyarrow_array_to_numpy_and_mask(np_dtype_to_arrays):
+    """
+    Test conversion from pyarrow array to numpy array.
+
+    Modifies the pyarrow buffer to contain padding and offset, which are
+    considered valid buffers by pyarrow.
+
+    Also tests empty pyarrow arrays with non empty buffers.
+    See https://github.com/pandas-dev/pandas/issues/40896
+    """
+    from pandas.core.arrays._arrow_utils import pyarrow_array_to_numpy_and_mask
```

Comment:
can you import at the top (after the pyarrow check is fine)

Reply:
Should I move the rest of the pandas imports below the pyarrow check as well? Looks a bit weird having them seperated by the check.